language/move-stdlib/tests/BCSTests.move

```diff
@@ -0,0 +1,95 @@
+#[test_only]
+module 0x1::BCSTests {
+    use 0x1::BCS;
+
+    struct Box<T> has copy, drop, store { x: T }
+    struct Box3<T> has copy, drop, store { x: Box<Box<T>> }
+    struct Box7<T> has copy, drop, store { x: Box3<Box3<T>> }
+    struct Box15<T> has copy, drop, store { x: Box7<Box7<T>> }
+    struct Box31<T> has copy, drop, store { x: Box15<Box15<T>> }
+    struct Box63<T> has copy, drop, store { x: Box31<Box31<T>> }
+    struct Box127<T> has copy, drop, store { x: Box63<Box63<T>> }
+    struct Box255<T> has copy, drop, store { x: Box127<Box127<T>> }
+
+    #[test]
+    fun bcs_address() {
+        let addr = @0x89b9f9d1fadc027cf9532d6f99041522;
+        let expected_output = x"89b9f9d1fadc027cf9532d6f99041522";
+        assert(BCS::to_bytes(&addr) == expected_output, 0);
+    }
+
+    #[test]
+    fun bcs_bool() {
+        let expected_output = x"01";
+        assert(BCS::to_bytes(&true) == expected_output, 0);
+    }
+
+    #[test]
+    fun bcs_u8() {
+        let expected_output = x"01";
+        assert(BCS::to_bytes(&1u8) == expected_output, 0);
+    }
+
+    #[test]
+    fun bcs_u64() {
+        let expected_output = x"0100000000000000";
+        assert(BCS::to_bytes(&1) == expected_output, 0);
+    }
+
+    #[test]
+    fun bcs_u128() {
+        let expected_output = x"01000000000000000000000000000000";
+        assert(BCS::to_bytes(&1u128) == expected_output, 0);
+    }
+
+    #[test]
+    fun bcs_vec_u8() {
```

Comment:
could probably use some more tests for simple structs and vectors but that can be another PR

Reply:
Agreed. This is just a pretty straightforward port. But there are definitely other tests we'll want to add -- I'll probably add those in next week sometime.